### PR TITLE
FLCRM-12761 Add support for geometry

### PR DESCRIPTION
**Pull request by toddrun**
# What?

- Makes `record.js` accept more geometry types than simple Points
- Enables clients to set LineString or Polygon GeoJSON objects

The diff makes it look like a _lot_ was changed. In reality, the `src/records.js` and a couple of fixtures used within a test are the only non-distributable files changed.

# Why?

https://fulcrumapp.atlassian.net/browse/FLCRM-12761

Needed for Lines and Polygon in web where we need to pass a GeoJSON `geometry` object instead of just a lat/long pair.

# Testing

- I'm testing this in my namespace, using `"fulcrum-core": "fulcrumapp/fulcrum-core#FLCRM-12761-support-for-geometry",` within fulcrum components package.json file. 
- Editing location, as well as geometry, works as before

- I still need to do this: `NOTE TO AUTHOR: Notify Slack channel #chaos-env-updates of upcoming changes to this widely-used tool.`

### CODE REVIEW

File: dist/validation/date-format-validation-error.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
 export default class DateFormatValidationError extends ElementValidationError {
-    get message(): string;
+    get message(): any;
```
Comment (chrisworkmode) [inline dist/validation/date-format-validation-error.d.ts:2]: Why does this need to be changed to `any`?
Ideally, we keep it as specific of a type as possible. Having to code around `any` feels like a lot of work and potentially increases the chance of an error.
Same with the other messages functions below.

File: src/record.js [hunk 2]
```diff
@@ -425,16 +439,36 @@ export default class Record extends Feature {
   }
 
   get geometryAsGeoJSON() {
```
Comment (chrisworkmode) [inline src/record.js:441]: Can this be simplified to (see below)?
```
if (this.isGeometryEnabled && this.geometry) {
  return this.geometry;
}

if (this.hasCoordinate) {
  return this.buildPointFromLatLong();
}

return null;
```

File: dist/validation/date-format-validation-error.d.ts (hunk 1 above)
Comment (toddrun) [reply dist/validation/date-format-validation-error.d.ts:2]: That's in a dist file . . . I don't know how or why that changed. But I made a _lot_ of little incremental changes. Let me see if I can clean up the changes to the dist files.

Comment (toddrun) [reply dist/validation/date-format-validation-error.d.ts:2]: I can manually change that file, but the next person who does a `yarn build` is going to stomp on it.

File: src/record.js (hunk 2 above)
Comment (toddrun) [reply src/record.js:441]: Thanks. Much nicer!

File: dist/validation/date-format-validation-error.d.ts (hunk 1 above)
Comment (chrisworkmode) [reply dist/validation/date-format-validation-error.d.ts:2]: Oh strange it's changed in a dist file. I wonder why the transpiler things it should be any now. 😟

**Review by chrisworkmode — APPROVED** [review]

**Review by JuanAcosta112 — APPROVED** [review]

File: src/record.js [hunk 3]
```diff
@@ -425,16 +439,24 @@ export default class Record extends Feature {
...
-    if (!this.hasCoordinate) {
-      return null;
+    if (this.isGeometryEnabled && this.geometry) {
```
Comment (zhm) [inline src/record.js:442]: I don't think we need to check `isGeometryEnabled` here, it wasn't checking it before to return the `latitude` & `longitude`. `isGeometryEnabled` is not a new thing, it's just the "is location enabled?" setting, but I don't think we NEED to assert that here. Neither of the mobile apps have this check either for the simple accessor on a record.

**Review by zhm — APPROVED** [review]

**Pull request merged**